src/mpi/group/grouputil.c

```diff
@@ -156,28 +156,29 @@ int MPIR_Group_create(int nproc, MPIR_Group ** new_group_ptr)
 int MPIR_Group_dup(MPIR_Group * old_group, MPIR_Session * session_ptr, MPIR_Group ** new_group_ptr)
 {
     int mpi_errno = MPI_SUCCESS;
-    MPIR_Group *new_group;
 
-    new_group = (MPIR_Group *) MPIR_Handle_obj_alloc(&MPIR_Group_mem);
-    if (!new_group) {
-        mpi_errno = MPIR_Err_create_code(MPI_SUCCESS, MPIR_ERR_RECOVERABLE, "MPIR_Group_dup",
-                                         __LINE__, MPI_ERR_OTHER, "**nomem", 0);
-        goto fn_fail;
-    }
-    MPIR_Object_set_ref(new_group, 1);
+    *new_group_ptr = (MPIR_Group *) MPIR_Handle_obj_alloc(&MPIR_Group_mem);
+    MPIR_ERR_CHKANDJUMP(!*new_group_ptr, mpi_errno, MPI_ERR_OTHER, "**nomem");
+    MPIR_Object_set_ref(*new_group_ptr, 1);
+
+    (*new_group_ptr)->size = old_group->size;
+    (*new_group_ptr)->rank = old_group->rank;
+    MPIR_Group_set_session_ptr(*new_group_ptr, session_ptr);
+    memcpy(&(*new_group_ptr)->pmap, &old_group->pmap, sizeof(struct MPIR_Pmap));
```

Comment:
Related to comment on map reusing  in PR #7235, should we add refcount to `MPIR_Pmap` so that it can be easily shared in most of the scenario. This would reduce memory usage due to these linear data structures.

Reply:
`MPIR_Pmap` is never shared and belong to an `MPIR_Group`. `MPIR_Group`, on the other hand, can be shared by comms or psets, and they are reference-counted.